waku/v2/node/peer_manager/peer_manager.nim

```diff
@@ -386,6 +394,41 @@ proc relayConnectivityLoop*(pm: PeerManager) {.async.} =
 
     await sleepAsync(ConnectivityLoopInterval)
 
+proc prunePeerStore(pm: PeerManager) {.async.}  =
+  while pm.started:
+    let numPeers = toSeq(pm.peerStore[AddressBook].book.keys).len
+    let capacity = pm.peerStore.capacity
+    if numPeers < capacity:
+      await sleepAsync(PrunePeerStoreInterval)
+      continue
```

Comment:
Split things into sub-procedures so you can `return` instead of repeating the `sleepAsync(...)` code. That will make the code more readable.

The sub-procedure should be a linear function wrapped and called by the periodic execution function. Mixing the loop code with the linear code makes things more complex to understand.

Reply:
indeed! should be fixed here. 3df173c6b77edf7bcfd049ed197dffc675bccede
makes it easier to unit test, added one in a94325ab694bb38d34e655a7ba01a68dec76600b